src/app/donation-start/donation-start-form/donation-start-form.component.ts

```diff
@@ -642,37 +649,11 @@ export class DonationStartFormComponent implements AfterContentChecked, AfterCon
     // Jump back if we get an out of band message back that the card is *not* valid/ready.
     // Don't jump forward when the card *is* valid, as the donor might have been
     // intending to edit something else in the `payment` step; let them click Next.
-    if (!this.donation || !this.stripePaymentMethodReady || !this.card) {
+    if (!this.donation || !this.stripePaymentMethodReady || !this.stripePaymentElement || !this.stripeElements) {
       this.jumpToStep('Payment details');
 
       return;
     }
-
```

Comment:
I'm not sure if/why we needed to explicitly create stripe payment
    methods. Stripe says that:
    
            In most integrations, you will not need to use this method [createPaymentMethod].
            Instead, use methods like stripe.confirmCardPayment, which
            will automatically create a PaymentMethod when you confirm a PaymentIntent.
            https://stripe.com/docs/js/payment_methods/create_payment_method
    
    I tried calling createPaymentMethod by passing the elements instance,
    but it gives an error saying you have to call elements.submit first,
    which I think we don't want to do before the donor presses the pay now
    button.

Reply:
@bdsl This sounds very closely related to what you mentioned about collecting billing address info like `country`. I think that if we kept this as it was, we could hide the address fields.

This lets us provide a smoother journey because we can pre-fill country from CDN header and postcode from the Gift Aid step.

Reply:
As discussed, the docs strongly suggest `submit()` is something we can/should do around this point in the journey. So something close to what we had before seems right.

Reply:
Looks like trying to manually create the payment method if we don't allow the payment element to collect billing details has the same problem as trying to confirm the payment directly without - we get errors asking us to send stripe info we don't have, like donor phone number etc:

![image](https://github.com/thebiggive/donate-frontend/assets/159481/bbfe009a-e70a-4f09-bb94-95dd93c9d391)


[diff.txt](https://github.com/thebiggive/donate-frontend/files/12513362/diff.txt)

Reply:
We could configure the payment element to not ask for country and postcode, but again if we try to create the payment method as soon as its filled in we have to pass these, and the way we've arranged the form we wouldn't expect them to be filled in until *after* the payment element part:

![image](https://github.com/thebiggive/donate-frontend/assets/159481/800b2be7-029d-4768-9415-0279d14c85bb)

The country is filled but just with a default value that won't generally be accurate.

![image](https://github.com/thebiggive/donate-frontend/assets/159481/be01f6a1-56c3-4294-8d25-6ba2979b8fa5)

Reply:
Yes we could get country from the header, although that seems odd when we already have country drop-down. We can't rely on the gift-aid step for postcode since that only asks for postcode if gift aid is turned on.